core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -1511,6 +1626,18 @@ private void handleErrorInSyncContext(Status error) {
       scheduleExponentialBackOffInSyncContext();
     }
 
+    // Must run in SynchronizationContext.
+    private void drainPendingCalls() {
+      if (pendingCalls == null) {
+        return;
+      }
+      for (RealChannel.PendingCall<?, ?> pendingCall : pendingCalls) {
+        pendingCall.pendingCallRunnable.run();
+      }
+      inUseStateAggregator.updateObjectInUse(pendingCallsInUseObject, false);
```

Comment:
I think you actually had this correct before. We need to wait until the calls are drained (at least the newCall and start() for each call) until we consider this no longer used, as otherwise that opens up a race window where it appears there are no RPCs in progress.

Reply:
Yeah, that's subtle. Fixed.